src/uk/zenko/src/eu/kanade/tachiyomi/extension/uk/zenko/Zenko.kt

```diff
@@ -0,0 +1,199 @@
+    override fun chapterListRequest(manga: SManga): Request {
+        val mangaId = manga.url.substringAfterLast('/')
+        val url = "$API_URL/titles/$mangaId/chapters"
+        return GET(url, headers)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val result = response.asClass<List<ChapterResponseItem>>()
+        return result.sortedByDescending { item ->
+            val id = StringProcessor.generateId(item.name)
+            if (id > 0) id else item.id.toDouble()
+        }.map { chapterResponseItem ->
+            SChapter.create().apply {
+                url = "$baseUrl/titles/${chapterResponseItem.titleId}/${chapterResponseItem.id}"
+                name = StringProcessor.format(chapterResponseItem.name)
+                date_upload = chapterResponseItem.createdAt!!.secToMs()
+                scanlator = chapterResponseItem.publisher!!.name
+            }
+        }
+    }
+
+    // =============================== Pages ================================
+    override fun pageListRequest(chapter: SChapter): Request {
+        val chapterId = chapter.url.substringAfterLast('/')
+        val url = "$API_URL/chapters/$chapterId"
+        return GET(url, headers)
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        val data = response.asClass<ChapterResponseItem>()
+        return data.pages!!.map { page ->
+            Page(page.id, "", "$IMAGE_STORAGE_URL/${page.imgUrl}")
+        }
+    }
+
+    override fun imageUrlParse(response: Response): String = ""
+
+    // ============================= Utilities ==============================
+    private fun parseAsMangaResponseDto(response: Response): MangasPage {
+        val zenkoMangaListResponse = response.asClass<ZenkoMangaListResponse>()
+        return makeMangasPage(zenkoMangaListResponse.data, zenkoMangaListResponse.meta.hasNextPage)
+    }
+
+    private fun offsetCounter(page: Int): Int {
+        var currentOfsset = 0
+        val offset = if (page == 1) {
+            currentOfsset = 0 // for page 1 offset must be 0
+            currentOfsset
+        } else {
+            currentOfsset = (page - 1) * 15 // calculate offset for other pages
+            currentOfsset
+        }
+        return offset
+    }
+
+    private fun makeZenkoMangaRequest(offset: Int, sortBy: String): Request {
+        val url = "$API_URL/titles".toHttpUrl().newBuilder()
+            .addQueryParameter("limit", "15")
+            .addQueryParameter("offset", "$offset")
+            .addQueryParameter("sortBy", sortBy)
+            .addQueryParameter("order", "DESC")
+            .build()
+        return GET(url, headers)
+    }
+
+    private fun makeMangasPage(
+        mangaList: List<MangaDetailsResponse>?,
+        hasNextPage: Boolean = false,
+    ): MangasPage {
+        return MangasPage(
+            mangaList!!.map(::makeSManga),
+            hasNextPage,
+        )
+    }
```

Comment:
```suggestion
    private fun makeMangasPage(
        mangaList: List<MangaDetailsResponse>,
        hasNextPage: Boolean = false,
    ): MangasPage {
        return MangasPage(
            mangaList.map(::makeSManga),
            hasNextPage,
        )
    }
```
make the dto not nullable since you are adding non null assertion anyway